.github/workflows/docker-image.yml

```
@@ -0,0 +1,78 @@
+name: Docker Image CI
+
+#This only pushes to Docker Hub at the moment. To push to GitHub Packages, other steps are needed: https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+    inputs:
+      versionTag:
+        description: 'Version tag to push to Docker Hub (lowercase, alphanumeric)'
+        required: true
+        type: string
+        
+
+jobs:
+
+  build_test_push:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false # So that remaining jobs don't instantly quit if one fails
+      matrix:
+        include: # Platform locates Dockerfile ("Dockerfile-{PLATFORM}"), docker tag has to be all lowercase alphanumeric for mysterious docker reasons
+          - platform: CUDA10.2
+            dockertag: cuda102
+          - platform: CUDA11.3
+            dockertag: cuda113
+          - platform: CPU
+            dockertag: cpu
+          - platform: ROCm
+            dockertag: rocm
+
+    steps:
+    - name: Check out the repository
+      uses: actions/checkout@v2
+      with:
+        lfs: true
+        submodules: 'recursive'
+        
+    - name: Log in to Docker Hub
+      uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
```

```suggestion
      uses: docker/login-action@v1
```
Might this be the issue?

---

It's actually because of built-in security settings on GitHub. A PR from a fork can't access your repo's secrets, even if they exist, unless a special trigger is used that only runs the target branch workflows (so I can't just make my own action, "Print All Secrets And Do Bad Things" and run it on your repo). But then that target branch wouldn't have this workflow, so it wouldn't work. 🤷‍♂️ It's a little complicated.

It would work once merged, or if we pushed this to a non-master branch of this repo and then opened a PR to master from there instead. Up to you. I'll change it to use v1 anyway though.

---

Ah, understood. Let me push a new branch for this.